component/componenttest/example_factories.go

```diff
@@ -163,31 +164,8 @@ var exampleReceivers = map[configmodels.Receiver]*ExampleReceiverProducer{}
 // MultiProtoReceiver is for testing purposes. We are defining an example multi protocol
 // config and factory for "multireceiver" receiver type.
 type MultiProtoReceiver struct {
-	TypeVal   configmodels.Type                   `mapstructure:"-"`
-	NameVal   string                              `mapstructure:"-"`
-	Protocols map[string]MultiProtoReceiverOneCfg `mapstructure:"protocols"`
-}
-
-var _ configmodels.Receiver = (*MultiProtoReceiver)(nil)
-
-// Name gets the exporter name.
-func (rs *MultiProtoReceiver) Name() string {
-	return rs.NameVal
-}
-
-// SetName sets the receiver name.
-func (rs *MultiProtoReceiver) SetName(name string) {
-	rs.NameVal = name
-}
-
-// Type sets the receiver type.
-func (rs *MultiProtoReceiver) Type() configmodels.Type {
-	return rs.TypeVal
-}
-
-// SetType sets the receiver type.
-func (rs *MultiProtoReceiver) SetType(typeStr configmodels.Type) {
-	rs.TypeVal = typeStr
+	configmodels.ReceiverSettings `mapstructure:",squash"`            // squash ensures fields are correctly decoded in embedded struct
+	Protocols                     map[string]MultiProtoReceiverOneCfg `mapstructure:"protocols"`
```

Comment:
Indeed, was changing the unmarshal functions around and saw this. I can split it.

Reply:
Split in https://github.com/open-telemetry/opentelemetry-collector/pull/1476, once that is merged this will be no-op. I will not revert for the moment with the idea that the other PR is merged first so just rebasing removes this change.